DoSomething/gladiator
# Add key authorization for post requests to Gladiator.

#### What's this PR do?
This PR updates the API to add a middleware for authorizing specific request methods. This update adds a conditional for `POST` requests that requires supplying a `X-DS-Gladiator-API-Key` in the header of the request with a specified key as the value. 

If the key matches what is on Gladiator then the request is accepted, otherwise it is rejected and responds with an error, which is caught by the `Handler` class and responds with a formatted JSON API error.

The exception thrown is a new `UnauthorizedAccessException` which could be utilized for other conditions in the web app as well regarding unauthorized access to data. 

#### How should this be manually tested?
Pull down branch, `composer dump-autoload`, make sure to add the new `GLADIATOR_API_KEY=uxm107oct1977` to your **.env** file and then try making a POST request to the api with a valid api key, and then with an invalid one. Does the API respond as expected and described above?

#### Any background context you want to provide?
There's a new env variable that needs to be added to prod, qa, and local env for the API key:

```
GLADIATOR_API_KEY=uxm107oct1977
```

#### What are the relevant tickets?
Fixes #69


---
@angaither @DFurnes @sbsmith86 @deadlybutter
